gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/detached/DetachedVertex.java

```diff
@@ -74,6 +75,38 @@ protected DetachedVertex(final Vertex vertex, final boolean withProperties) {
         }
     }
 
+    protected DetachedVertex(final Vertex vertex,
+                             final DetachStrategy.DetachOptions detachOptions) {
+        super(vertex);
+
+        if (detachOptions.getDetachMode() == DetachStrategy.DetachMode.NONE
+                || detachOptions.getDetachMode() == DetachStrategy.DetachMode.CUSTOM && detachOptions.getProperties() == null)
+            return;
+
+        final Iterator<VertexProperty<Object>> propertyIterator = vertex.properties();
+        if (!propertyIterator.hasNext())
+            return;
+
+        this.properties = new HashMap<>();
+
+        if (detachOptions.getDetachMode() == DetachStrategy.DetachMode.ALL) {
```

Comment:
There is some code duplication between this case and DetachedVertex(vertex, true). Could potentially have the above constructor (DetachedVertex(final Vertex vertex, final boolean withProperties)) make calls to this constructor to avoid duplication.

Reply:
fixed in https://github.com/apache/tinkerpop/pull/1843/commits/8290b2049fb4876fd7ddeafd18a2345222f85597